hybrasyl/Objects/User.cs

```diff
@@ -2748,11 +2748,50 @@ public void AssailAttack(Xml.Direction direction, Creature target = null)
                     //UseCastable(c, target);
                 }
             }
+            var motionId = (byte)1;
             //animation handled here as to not repeatedly send assails.
-            var firstAssail = SkillBook.FirstOrDefault(x => x.Castable.IsAssail);
-            var motion = firstAssail?.Castable?.Effects.Animations.OnCast.Player.FirstOrDefault(y => y.Class.Contains(Class));
+            if(Class == Xml.Class.Warrior)
+            {
+                if(SkillBook.Any(b => b.Castable.Name == "Wield Two-Handed Weapon"))
+                {
+                    if (Equipment.Weapon != null && Equipment.Armor != null)

```

Comment:
i think this is more concisely expressed as
`if (Equipment.Weapon?.WeaponType == WeaponType.TwoHand && Equipment.Armor?.Class == Xml.Class.Warrior)`

Reply:
tried this to begin with, null check fails unless broken apart.